packages/apollo-engine-reporting/src/schemaReporter.ts

```diff
@@ -139,14 +156,29 @@ export class SchemaReporter {
           'https://engine.apollographql.com/ to obtain an API key for a service.',
         ].join(' '),
       );
-    } else if (data.me.__typename === 'ServiceMutation') {
-      if (!data.me.reportServerInfo) {
-        throw new Error(msgForUnexpectedResponse(data));
+    } else if (
+      data.me.__typename === 'ServiceMutation' &&
+      data.me.reportServerInfo
+    ) {
+      if (data.me.reportServerInfo.__typename == 'ReportServerInfoResponse') {
+        return { ...data.me.reportServerInfo, stopReporting: false };
+      } else {
+        this.logger.error(
+          [
+            'Received input validation error from Graph Manager:',
+            data.me.reportServerInfo.message,
+            'Stopping reporting. Please fix the input errors.',
+          ].join(' '),
+        );
+        this.stop();
+        return {
+          stopReporting: true,
+          inSeconds: 100,
```

Comment:
Why have a value here at all?

Reply:
I would as soon have a discriminated union for the response with either the fields of `inSeconds` and `withSchema` or `stopReporting: true`, but 🤷